ext/drutil/drutil.c

```diff
@@ -439,6 +465,43 @@ drutil_opnd_mem_size_in_bytes(opnd_t memref, instr_t *inst)
         uint sz = opnd_size_in_bytes(opnd_get_size(instr_get_dst(inst, 1)));
         ASSERT(opnd_is_immed_int(instr_get_src(inst, 1)), "malformed OP_enter");
         return sz * extra_pushes;
+    } else if (inst != NULL && instr_is_xsave(inst)) {
+        /*
+         * The following is an incomplete computation of the xsave instruction's
+         * written xsave area's size. Specifically, it
+         *
+         * - Ignores the user state mask components set in edx:eax, because it is
+         *   a dynamic value at runtime. The real output size of xsave depends on
+         *   the instruction's user state mask AND the user state mask as supported
+         *   by the CPU based on XCR0.
+         * - Ignores supervisor state component PT (bit 8).
+         * - Ignores the user state component PKRU state (bit 9).
+         * - Ignores the xsaveopt flavor of xsave.
+         * - Ignores the xsavec flavor of xsave (compacted format).
+         *
+         * It computes the expected size for the standard format of the x87 user
+         * state component (bit 0), the SSE user state component (bit 1), the AVX
+         * user state component (bit 2), the MPX user state components (bit 2 and 3)
+         * and the AVX-512 user state component (bit 7).
+         *
+         */
+#    ifdef WINDOWS
+        /* XXX i#2946: support Windows */
+#    else
+        /* XXX: we may want to re-factor and move functions like this into drx. */
```

Comment:
Maybe using the pre-existing versions in clients/drcpusim/tests/cpuid.c

Reply:
Added comment.